lib/elixir/lib/gen_event.ex

```diff
@@ -380,8 +380,21 @@ defmodule GenEvent do
         :gen.start(GenEvent, mode, @no_callback, [], [])
       atom when is_atom(atom) ->
         :gen.start(GenEvent, mode, {:local, atom}, @no_callback, [], [])
-      other when is_tuple(other) ->
-        :gen.start(GenEvent, mode, other, @no_callback, [], [])
+      {:global, _term} = tuple ->
+        :gen.start(GenEvent, mode, tuple, @no_callback, [], [])
+      {:via, module, _term} = tuple when is_atom(module) ->
+        :gen.start(GenEvent, mode, tuple, @no_callback, [], [])
+      other ->
+        raise ArgumentError, String.trim_trailing("""
+          expected :name option to be one of:
+
+            * nil
+            * atom
+            * {:global, term}
+            * {:via, module, term}
+
+          Got: #{inspect(other)}
+          """)
```

Comment:
Yes, I think indenting that list two spaces is fine :)

On Tuesday, 12 July 2016, alecs popa <notifications@github.com> wrote:

> In lib/elixir/lib/gen_event.ex
> <https://github.com/elixir-lang/elixir/pull/4994#discussion_r70458119>:
>
> > -        :gen.start(GenEvent, mode, other, @no_callback, [], [])
> > +      {:global, _term} = tuple ->
> > +        :gen.start(GenEvent, mode, tuple, @no_callback, [], [])
> > +      {:via, module, _term} = tuple when is_atom(module) ->
> > +        :gen.start(GenEvent, mode, tuple, @no_callback, [], [])
> > +      other ->
> > +        raise ArgumentError, String.trim_trailing("""
> > +          expected :name option to be one of:
> > +
> > +            * nil
> > +            * atom
> > +            * {:global, term}
> > +            * {:via, module, term}
> > +
> > +          Got: #{inspect(other)}
> > +          """)
>
> Thanks, I like following standards I just didn't look hard enough to find
> an example :(
> Can the
>
>   * nil
>   * atom
>   * {:global, term}
>   * {:via, module, term}
>
> be indented one level for readability?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/4994/files/3cb3c861f3b49e383d7b3e6c50777a28af5fea06#r70458119>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADtcSimixPgIAGKg4YZKv9KlK3xgoW6Mks5qU65TgaJpZM4JJ5O8>
> .
>


-- 

Andrea Leopardi
an.leopardi@gmail.com